models/things.js

```diff
@@ -116,7 +116,7 @@ var Things = {
 
      // Notify each open websocket of the new Thing
      this.websockets.forEach(function(socket) {
-       socket.send(JSON.stringify(thing.getDescription()));
+       socket.send(JSON.stringify(thing));
```

Comment:
There's a difference between a "new thing" and a "thing". New things come from the adapter manager and get their thing description from Device.asThing(). Things come from the database and get their description from Thing.getDescription().

New things have an internal ID in their thing description until they are turned into things in the database which have a URL to identify them in the web API. Things using the Thing Description format we're aiming to standardise do not have an ID property because their URL is their ID.

What's happening here is the handleNewThing() is using the Thing constructor to validate the description of the new thing, which ends up stripping out the internal ID. I don't actually know how this is working because the client side needs the ID to generate the POST request https://github.com/moziot/gateway/blob/master/static/js/add-thing.js#L115 (it might be that you've found a bug).

I'd suggest that either you remove the call to the Thing() constructor in handleNewThing and pass the new thing description (including its id) directly to the websocket message as is done elsewhere, or you simply add back in the ID after it has been validated.

It's good to be getting extra eyes on this code to find these inconsistencies :)